database/database.go

```diff
@@ -0,0 +1,304 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/NebulousLabs/skynet-accounts/build"
+	"github.com/NebulousLabs/skynet-accounts/user"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	// EnvDBHost holds the name of the environment variable for DB host.
+	EnvDBHost = "SKYNET_DB_HOST"
+	// EnvDBPort holds the name of the environment variable for DB port.
+	EnvDBPort = "SKYNET_DB_PORT"
+	// EnvDBUser holds the name of the environment variable for DB username.
+	EnvDBUser = "SKYNET_DB_USER"
+	// EnvDBPass holds the name of the environment variable for DB password.
+	EnvDBPass = "SKYNET_DB_PASS"
+
+	// mongoCompressors defines the compressors we are going to use for the
+	// connection to MongoDB
+	mongoCompressors = "zstd,zlib,snappy"
+	// mongoReadPreference defines the DB's read preference. The options are:
+	// primary, primaryPreferred, secondary, secondaryPreferred, nearest.
+	// See https://docs.mongodb.com/manual/core/read-preference/
+	mongoReadPreference = "nearest"
+	// mongoWriteConcern describes the level of acknowledgment requested from
+	// MongoDB.
+	mongoWriteConcern = "majority"
+	// mongoWriteConcernTimeout specifies a time limit, in milliseconds, for
+	// the write concern to be satisfied.
+	mongoWriteConcernTimeout = "1000"
+
+	// DBName defines the name of Skynet's database.
+	DBName = "skynet"
+	// DBUsersCollection defines the name of the "users" collection within
+	// skynet's database.
+	DBUsersCollection = "users"
+
+	// ErrUserNotFound is returned when we can't find the user in question.
+	ErrUserNotFound = errors.New("user not found")
+	// ErrEmailAlreadyUsed is returned when we try to use an email to either
+	// create or update a user and another user already uses this email.
+	ErrEmailAlreadyUsed = errors.New("email already in use by another user")
+	// ErrGeneralInternalFailure is returned when we do not want to disclose
+	// what kind of error occurred. This should always be coupled with another
+	// error output for internal use.
+	ErrGeneralInternalFailure = errors.New("general internal failure")
+
+	// True is a helper, so we can easily provide a *bool for UpdateOptions.
+	True = true
+	// False is a helper, so we can easily provide a *bool for UpdateOptions.
+	False = false
```

Comment:
These are not even used right?

Reply:
I was using one of them. We will be using them in the future.

Reply:
I'll just remove them for the moment. YAGNI.